src/registers/control.rs

```diff
@@ -245,7 +248,13 @@ mod x86_64 {
     impl Cr2 {
         /// Read the current page fault linear address from the CR2 register.
         #[inline]
-        pub fn read() -> VirtAddr {
+        pub fn read() -> Result<VirtAddr, VirtAddrNotValid> {
+            VirtAddr::try_new(Self::read_raw())
```

Comment:
Not related to this PR, but I think we should make the wrapped `u64` in `VirtAddrNotValid` a public field.

Reply:
Agreed, I opened #340 to add this.